substrate/frame/staking/src/pallet/mod.rs

```diff
@@ -1315,9 +1307,7 @@ pub mod pallet {
 				Error::<T>::ControllerDeprecated
 			);
 
-			let _ = ledger
-				.set_payee(payee)
-				.defensive_proof("ledger was retrieved from storage, thus its bonded; qed.")?;
+			ledger.set_payee(payee)?;
```

Comment:
It can fail for other reasons than unbonded (not exists in storage).

Reply:
The current assumptions are:
1. If ledger is bonded, then a new ledger is returned from `Self::ledger`;
2. If ledger is bonded, then it has a Payee entry 

What changed in these assumptions with this PR? `ledger.set_payee(payee)` should always be Ok in this context (provided there's no underlying issues with the storage state, as we've seen recently. But in this case, it's a system bug and should not happen).

Reply:
I agree with @gpestana , I don't think there's anything that changed regarding those assumptions (anymore). The previous impl had some restricted rewards destination checks IIRC, but they are no longer there so we're back to the initial assumptions.